### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by nasearle**
This commit adds an authentication demo app in the demo folder. The app
is a mock e-commerce app that requires the user to sign in to purchase
an item. The app demos a client-side Google sign in. It also contains a
straw-man custom sign-in that doesn't actually work to motivate the
Google sign in.

### CODE REVIEW

**Comment by DavidScales** [comment]
Can you update the README to include the following:

```
## Note

This is not an official Google product.

All images are are licensed CC0 - no attribution required.
```
and possibly update it more generally? For reference the other demo's README is:

```
# Debugging Service Workers demo

This sample application can be used to demonstrate techniques for debugging
service workers.

This content is part of Google's [Progressive Web App Training](https://developers.google.com/web/ilt/pwa/),
which is also available on [GitBook](https://www.gitbook.com/book/google-developer-training/progressive-web-apps-ilt-codelabs/details). 

## Note

This is not an official Google product.

All images are are licensed CC0 - no attribution required.
```

File: demos/authentication/app/server.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+app.get('/pages/signup', (req, res) => {
+  res.sendFile(__dirname + '/pages/signup.html');
+});
+
+app.get('/pages/signin', (req, res) => {
+  res.sendFile(__dirname + '/pages/signin.html');
+});
+
+app.post('/submit-signin', (req, res) => {
+  // TODO - send message like "Uh-oh something went wrong"
```
Comment (DavidScales) [inline demos/authentication/app/server.js]: this TODO. also exists in the solution

**Comment by DavidScales** [comment]
also Chrome warns me that there is no autocomplete attribute on the password field:

`[DOM] Input elements should have autocomplete attributes (suggested: "current-password"): (More info: https://goo.gl/9p2vKq) `

**Comment by nasearle** [comment]
Added in your suggestions

**Pull request merged**